File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& a) {
        map<long long, long long> mp;
        int n = a.size();

        sort(a.begin(), a.end());
        long long ans = 0;

        for (int i = 0; i < n; ++i) {
            if (i == 0)
                mp[a[i]] = 1;
            else
                mp[a[i]] = max(1LL, mp[a[i] - 1] + 1);
            ans = max(ans, mp[a[i]]);
        }
        return ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
  if (nums.length == 0) return 0;
  nums.sort((a, b) => a - b);
  let ans = 0, cur = 1;

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] === nums[i - 1]) continue;
    if (nums[i] === nums[i - 1] + 1) {
      cur++;
    } else {
      ans = Math.max(ans, cur);
      cur = 1;
    }
  }
  return Math.max(ans, cur);
};